### PR TITLE
Convert quickstart from SSM agent to Bzero agent

### PR DESCRIPTION
**Pull request by chad-cwc**
## Description of the change

I updated QuickStart to install the Bzero agent instead of our SSM agent.

One major change here is that the SSM autodiscovery script returned the target ID, but the bzero autodiscovery was not written that way. The targets hostname, which should be the BastionZero target name, is being used instead. This isn't ideal, since Bzero targets aren't guaranteed to have unique names. To somewhat mitigate the risk of finding the wrong target based on name, the targets that exist prior to attempting the bzero install are saved and only targets NOT in that saved list are considered "new" for searching for a match.

Other fixes:
* fixed doc links
* fixed error message for when bzero is already installed - this was potentially misleading for cases where another error happened but was reported as "bzero is already installed"

## Testing

Describe how to test this PR....

**backend branch:** 
**bzero branch:** develop

#### Ready to run system tests?

- [x] Yes

## Relevant release note information

Release Notes: Transition QuickStart to install Bzero agent instead of SSM

## Related JIRA tickets

Relates to JIRA: CWC-2099

## Relevant HotFix if applicable

Relevant HotFix PR Number:

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [x] No

If yes, please explain:

### CODE REVIEW

File: src/services/quickstart/quickstart.service.ts [hunk 1]
```diff
@@ -109,23 +143,31 @@ export class QuickstartSsmService {
...
                 // Add fail message to current prefixText of the spinner and push to transcript
-                const errMsg = chalk.red(`BastionZero is already installed on SSH host: ${registrableHost.host.sshHost.name}. This target will be skipped.`);
-                this.consoleAndTranscript.pushToTranscript(errMsg);
-                spinner.prefixText = spinner.prefixText + errMsg + '\n';
+                let errorMessage;
+                if (error instanceof BzeroAlreadyInstalledError) {
+                    errorMessage = chalk.red(`BastionZero is already installed on SSH host: ${registrableHost.host.sshHost.name}. This target will be skipped.`);
+                } else {
+                    errorMessage = chalk.red(`✖ Failed to add SSH host: ${registrableHost.host.sshHost.name} to BastionZero. ${error}`);
+                }
```
Comment (ymarcus93) [inline src/services/quickstart/quickstart.service.ts:167]: Thanks for fixing this!

File: src/services/quickstart/quickstart.service.ts [hunk 2]
```diff
@@ -38,27 +48,51 @@ export class QuickstartSsmService {
...
+            };
+
+            try {
+                if (!targetId) {
+                    // Get all bzero targets and look for one with name matching bzeroTargetName.
+                    let bzeroTargets = await bzeroTargetHttpService.ListBzeroTargets();
+                    if (targetsToExclude) {
+                        // Remove exluded targets from new list of targets.
+                        bzeroTargets = bzeroTargets.filter(target => targetsToExclude.every(exludedTarget => exludedTarget.id !== target.id));
```
[the remark this answers is not in the document]
Comment (chad-cwc) [reply src/services/quickstart/quickstart.service.ts]: Removed as part of https://github.com/bastionzero/zli/pull/555/commits/cd432a59c1fef79febd9f3d3b04105cc79744835

[the remark this answers is not in the document]
Comment (chad-cwc) [reply src/services/quickstart/quickstart.service.ts]: https://github.com/bastionzero/zli/pull/555/commits/cd432a59c1fef79febd9f3d3b04105cc79744835

File: src/services/quickstart/quickstart.service.ts [hunk 3]
```diff
@@ -38,27 +48,51 @@ export class QuickstartSsmService {
...
 
-                const target = await ssmTargetHttpService.GetSsmTarget(ssmTargetId);
-                if (target.status === TargetStatus.Online && target.agentVersion !== '') {
-                    resolve(target);
+        const bzeroTargetHttpService = new BzeroTargetHttpService(this.configService, this.logger);
+        let targetId: string;
+
+        return retrier.resolve(() => new Promise<BzeroAgentSummary>(async (resolve, reject) => {
+            const checkTarget = (bzeroTarget: BzeroAgentSummary) => {
+                if (bzeroTarget.status === TargetStatus.Online && bzeroTarget.agentVersion !== '') {
```
[the remark this answers is not in the document]
Comment (chad-cwc) [reply src/services/quickstart/quickstart.service.ts]: https://github.com/bastionzero/zli/pull/555/commits/cd432a59c1fef79febd9f3d3b04105cc79744835

**Review by lipmas — APPROVED** [review]

**Review by lipmas — APPROVED** [review]

**Review by ymarcus93 — APPROVED** [review]

**Pull request merged**